tools/src/test/java/org/apache/pdfbox/tools/TestAddText.java

```diff
@@ -32,33 +35,123 @@ public void testLoadFileAndInitializeStream() throws IOException
+        str = str.replace("\n", "").replace("\r", "");
 
-            key = key.replace("\n", "").replace("\r", "");
+        String expected = key + " annotation";
 
-            actual.close();
+        expected = expected.replace("\n", "").replace("\r", "");
 
-            assertTrue(str.equals(key));
-        }
-        catch (IOException e)
+        Boolean result = str.equals(expected);
+
+        key = key.replace("\n", "").replace("\r", "");
+
+        add.writeText(0, key, PDType1Font.HELVETICA, 12);
+
+        actual.close();
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void testAnnotateTextBlankPage() throws IOException
+    {
+        AddText add = new AddText();
+        
+        String path = "../tools/src/main/java/org/apache/pdfbox/tools/AddTextResources/test_blank.pdf";
+
+        PDDocument actual = add.loadFileandInitializeStream(path);
+
+        PDFTextStripper stripper = new PDFTextStripper();
+        String key = stripper.getText(actual);
+        
+        add.annotateText(" annotation", 0);
+
+        String str = stripper.getText(actual);
+        str = str.replace("\n", "").replace("\r", "");
+
+        String expected = key + " annotation";
+
+        expected = expected.replace("\n", "").replace("\r", "");
+
+        Boolean result = str.equals(expected);
+
+        key = key.replace("\n", "").replace("\r", "");
+
+        add.writeText(0, key, PDType1Font.HELVETICA, 12);
+
+        actual.close();
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void testAnnotateTextMidPage() throws IOException
+    {
+        AddText add = new AddText();
+        
+        String old = "../tools/src/main/java/org/apache/pdfbox/tools/AddTextResources/test_2pages.pdf";
+        String path = "../tools/src/main/java/org/apache/pdfbox/tools/AddTextResources/test_2pages_new.pdf";
+
+        File in = new File(old);
+        File ou = new File(path);
+
+        try (FileInputStream fis = new FileInputStream(in);
+             FileOutputStream fos = new FileOutputStream(ou))
         {
-            assertEquals(1, 0);
+
+            byte[] buffer = new byte[1024];
+            int length;
+
+            while ((length = fis.read(buffer)) > 0) {
+
+                fos.write(buffer, 0, length);
+            }
         }
 
+        PDDocument actual = add.loadFileandInitializeStream(path);
+
+        PDFTextStripper stripper = new PDFTextStripper();
+        stripper.setStartPage(2);
+        stripper.setEndPage(2);
+        String key = stripper.getText(actual);
+        
+        add.annotateText(" annotation", 1);
+
+        String str = stripper.getText(actual);
+        str = str.replace("\n", "").replace("\r", "");
+
+        String expected = key.replace("\n", "").replace("\r", "");
+
+        expected = expected + " annotation";
+
+        Boolean result = str.equals(expected);
+
+        key = key.replace("\n", "").replace("\r", "");
+
+        add.writeText(1, key, PDType1Font.HELVETICA, 12);
```

Comment:
I think it would help to have each test try a different type of font to showcase this new utility.